pageserver/src/import_datadir.rs

```diff
@@ -192,19 +192,26 @@ fn import_rel<R: Repository, Reader: Read>(
     ensure!(len % pg_constants::BLCKSZ as usize == 0);
     let nblocks = len / pg_constants::BLCKSZ as usize;
 
-    if segno != 0 {
-        todo!();
-    }
-
     let rel = RelTag {
         spcnode: spcoid,
         dbnode: dboid,
         relnode,
         forknum,
     };
-    modification.put_rel_creation(rel, nblocks as u32)?;
 
     let mut blknum: u32 = segno * (1024 * 1024 * 1024 / pg_constants::BLCKSZ as u32);
+
+    // Call put_rel_creation for every segment of the relation,
+    // because there is no guarantee about the order in which we are processing segments.
+    // ignore "relation already exists" error
+    if let Err(e) = modification.put_rel_creation(rel, nblocks as u32) {
+        if e.to_string().contains("already exists") {
```

Comment:
Currently `put_rel_creation` errors if alredy exists, but `put_rel_extend` succeeds if it's already extended. A more consistent api would be nice. Maybe both of them return `bool`, whether any change was made? Or something similar to [insert](https://doc.rust-lang.org/stable/std/collections/struct.HashMap.html#method.insert).

Reply:
I am not sure, which API would be better here.
Maybe we can just introduce more informative error messages and handle them properly.
Let's leave it as TODO.